zerver/webhooks/slack/view.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message, create_stream_if_needed
+from zerver.lib.response import json_success, json_error
+from zerver.lib.validator import check_string, check_int
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.models import UserProfile, Client
+
+ZULIP_MESSAGE_TEMPLATE = "**{message_sender}** has sent the following message:\n `{text}`"
+
+@api_key_only_webhook_view('Slack')
+@has_request_variables
+def api_slack_webhook(request, user_profile, client,
+                      user_name=REQ(),
+                      text=REQ(),
+                      channel_name=REQ(),
+                      stream=REQ(default='slack'),
+                      channels_map_to_topics=REQ(default=1)):
+    # type: (HttpRequest, UserProfile, Client, str, str, str, str, int) -> HttpResponse
+
+    sender = user_name
+    message = text
+    channel = channel_name
+    channels_map_to_topics = int(channels_map_to_topics)
+
+    if channels_map_to_topics:
+        subject = "channel: {}".format(str(channel))
+        zulip_message = ZULIP_MESSAGE_TEMPLATE.format(message_sender=str(sender), text=str(message))
+        check_send_message(user_profile, client, "stream", [stream], subject, zulip_message)
+    else:
+        create_stream_if_needed(user_profile.realm, channel)
+        subject = _("message from Slack")
+        zulip_message = ZULIP_MESSAGE_TEMPLATE.format(message_sender=str(sender), text=str(message))
+        check_send_message(user_profile, client, "stream", [channel], subject, zulip_message)
```

Comment:
I think you can move it outside the `if...else...` statement, here just make quick "renaming"
`stream = channel`